### ISSUE TITLE
Feature request: Add a new bare backend

### ISSUE THREAD

**Issue by canatan** [swaywm/wlroots#1055]
I think having been a extremelly bare backend for this library is much usefull and wonderful.

Properties of such backed must include:

*  Single gpu, single connector, single encoder, single crtc, single mode(Always native or maximum resolution), explicit plane_ids(Not as an actual hw case but sw design choise)

* NO DEPENDENCİES other than wayland and mesa 

* 2 in 1.  Backend handles both input and gpu devices.

* Have only keyboard and pointer devices as input devices no more.

* Keyboard and pointers have to explicitly specified in command line as evdev node. 
eg rootson --keyboard=/dev/input/event4 --pointer=/dev/input/event7  

User(or Admin) of system have to guaranteed that non-root user able to read above nodes. It is up to them.

* The whole command line must be match below form. Everytings must be specified explicitly.

eg. rootson --keyboard=/dev/input/event4 --pointer=/dev/input/event7 --gpu=/dev/dri/card0 --render=/dev/dri/renderD128 --connector_id=19 encoder_id=43 --crtc_id=76 primary_plane_id=21 overlay_plane_ids=23,27,31

So when above case given "bare" backend choosen automatically.
User(or Admin) of system have to guaranteed that non-root user able to r/w above gpu node. It is up to them.

* No seat concepts. Always only one seat. Always only one output.

Such a backend will be exremelly useful, flexible, powerful especielly for raspery-pi like hardware.

It is true that my own try to implement this but no luck, i did't achieve.
It seems to me that everything tied everything, can't touch anything.
 
thoughts?

**Comment by emersion** [comment]
>Such a backend will be exremelly useful, flexible, powerful especielly for raspery-pi like hardware.

Why can't you use the DRM + libinput backends?

**Comment by canatan** [comment]
Certainly it is work but libinput pull a lot of dependency, such backend give me much power, much freedom, will be available for every platform, easy to implement, easy to maintenance, even if you wish cut it even more eg. drop thirty-party features(screen shooter, idle-inhibit ...), no floating windows to lightenig it:)

**Comment by emersion** [comment]
libinput doesn't have a lot of dependencies. Also, I don't want to re-implement libinput by reading directly from /dev.

You can already specify a DRM device when creating the DRM backend. If you just want to remove the libinput dependency, you could just implement a backend that reads from /dev if you want to.

Creating a backend that only works for this use-case will duplicate code already in the DRM and libinput backends and will be hard to maintain.

>2 in 1. Backend handles both input and gpu devices.

-1

>drop thirty-party features(screen shooter, idle-inhibit ...)

The backend has nothing to do with this.

>No seat concepts

Again, the backend doesn't decide how seats are managed.

All in all, it seems to me that you just want to create a bare compositor, not a bare backend.

**Comment by ascent12** [comment]
I don't want to duplicate a huge amount of the DRM code or possibly add extra layers of complication by allowing specifying DRM object ids. It's just more separate code paths that need to be tested and maintained, and I don't see the value in it.

By design, Wayland is dynamic by default, and we followed that in wlroots. By doing everything "right", we should be able to run well on systems ranging from high-end desktops with multiple discrete GPUs down to low-power ARM devices, all with the same code. Judging from the users we've gotten, I'd say we've achieved that.
I don't want to move away from that design just so we can shave off a couple of megabytes of memory usage.
